hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/HoodieCompactor.java

```diff
@@ -145,7 +145,7 @@ public HoodieData<WriteStatus> compact(
     if (useFileGroupReaderBasedCompaction) {
       ReaderContextFactory<T> readerContextFactory = context.getReaderContextFactory(metaClient);
       return context.parallelize(operations).map(
-              operation -> compact(compactionHandler, config, operation, compactionInstantTime, readerContextFactory.getContext()))
+              operation -> compact(compactionHandler, config, operation, compactionInstantTime, readerContextFactory.getContext(metaClient.getTableConfig())))
```

Comment:
When instantiating the `readerContextFactory` the meta client is passed in.  Should the reader context factory keep the table config and reuse that when `readerContextFactory.getContext` is invoked so there is no need to pass in `metaClient.getTableConfig()` again?

Reply:
Updated to use this approach